uint32 type
float32 goal_x
float32 goal_y
float32 goal_z